Tools/clinic/clinic.py

```diff
@@ -1988,13 +1988,38 @@ def parse_file(
     libclinic.write_file(output, cooked)
 
 
+def create_parser_namespace() -> dict[str, Any]:
```

Comment:
Can you use `functools.cache` instead of adding custom cache code? `functools` is already imported.

Reply:
Ok, done. I added `_create_parser_base_namespace()` function for that.

The function doesn't just return the same object at each call: each call creates a copy.